docs/KeyCaching.md

```diff
@@ -3,13 +3,16 @@
 Asherah supports caching of Intermediate and System Keys to minimize calls against external resource and reducing
 latency. This is especially important for minimizing calls to the configured
 [Key Management Service](KeyManagementService.md) as that is often the most expensive and latent interaction of the
-resources involved. The cache is built on top of the Secure Memory implementation.
+resources involved. The cache is built on top of the Secure Memory implementation. 
+The API also supports caching of Sessions, which further helps in reducing the latency by minimizing the number of API
+calls to the metastore. In case of stateless workloads, when the calling app can't make use of the shared session, the
+SDK makes use of the cached session instead of creating a new one. 
```

Comment:
Seems like this would read better if we reverse the order of statements. Something along the lines of:

`For stateless workloads, when the calling app can't make use of the shared session, Asherah also provide the option of caching sessions`. Then we can add the reducing latency part as well